File: monsoon3.0/src/main/java/wtf/monsoon/impl/module/movement/flight/MotionFlight.java
package wtf.monsoon.impl.module.movement.flight;

import io.github.nevalackin.homoBus.Listener;
import io.github.nevalackin.homoBus.annotations.EventLink;
import wtf.monsoon.api.module.Module;
import wtf.monsoon.api.setting.ModeProcessor;
import wtf.monsoon.api.setting.Setting;
import wtf.monsoon.impl.event.EventMove;
import wtf.monsoon.impl.event.EventPreMotion;

public class MotionFlight extends ModeProcessor {

    private final Setting<Double> speedValue = new Setting<Double>("Speed", 0.5)
            .minimum(0.0D)
            .maximum(2.0D)
            .incrementation(0.05D)
            .describedBy("The speed you will go.");

    public MotionFlight(Module parentModule) {
        super(parentModule);
    }

    @Override
    public void onDisable() {
        super.onDisable();
        mc.thePlayer.setVelocity(0, 0, 0);
    }

    @EventLink
    public final Listener<EventPreMotion> eventPreMotionListener = e -> {
        player.setSpeed(speedValue.getValue());
        mc.thePlayer.motionY = mc.gameSettings.keyBindJump.isKeyDown() ? speedValue.getValue() : mc.gameSettings.keyBindSneak.isKeyDown() ? -speedValue.getValue() : 0;
    };

    @Override
    public Setting[] getModeSettings() {
        return new Setting[] { speedValue };
    }

}

File: monsoon-3.x-v2/src/main/java/wtf/monsoon/client/modules/movement/flight/MotionFlight.kt
package wtf.monsoon.client.modules.movement.flight

import me.bush.eventbuskotlin.EventListener
import me.bush.eventbuskotlin.listener
import wtf.monsoon.backend.module.MulticlassModule
import wtf.monsoon.backend.module.mode.ModuleMode
import wtf.monsoon.backend.setting.Setting
import wtf.monsoon.client.event.EventMove
import wtf.monsoon.client.event.EventPacket
import wtf.monsoon.client.event.EventPreMotion
import wtf.monsoon.client.modules.exploit.disabler.NegativityDisabler
import wtf.monsoon.client.modules.movement.Flight

class MotionFlight(name: String, parent: MulticlassModule) : ModuleMode<Flight>(name, parent) {

    private val speed = Setting<Double>("Speed", "The speed of the flight.", 0.5, 0.05, 2.0, 0.05)
    private val timerSpeed = Setting<Float>("Timer Speed", "The timer speed of the flight.", 1.0f, 0.1f, 2.0f, 0.1f)

    init {
        this.registerSettings(speed, timerSpeed)
    }

    override fun disable() {
        super.disable()
        mc.timer.timerSpeed = 1.0f
    }

    @EventListener
    val moveEvent = fun(e: EventMove) {
        player.setSpeed(e, speed.getValue())
        e.y =
            if (mc.gameSettings.keyBindJump.isKeyDown) speed.getValue() else (if (mc.gameSettings.keyBindSneak.isKeyDown) -speed.getValue() else 0.0)
                .also { mc.thePlayer.motionY = it }

        mc.timer.timerSpeed = timerSpeed.getValue()
    }
}